docs/ActiveLabel.rst

```diff
@@ -1,135 +1,320 @@
+    # `label_module` acts as a filter to the `model_class` argument.
+    # Both `model_class` and `label_module` can be arrays
+    has_many :in, :human_actors, type: :ACTS_IN, label_module: :Actor, model_class: :Person
   end
 
+If you want more control over your association, you can use the ``node_labels:`` option instead
+
+.. code-block:: ruby
+
+  class Movie
+    include Neo4j::ActiveNode
+
+    # The node_labels option accepts a two dimentional array. Each array in the node_labels
+    # array includes a set of labels that the association will match against. In the example
+    # below, the ``actors`` association only includes nodes which have either ``:Actor:Person``
+    # OR ``:Actor:Animal`` labels and have an ``<-[:ACTS_IN]-`` relation to a ``Movie`` node
+    has_many :in, :actors, type: :ACTS_IN, node_labels: [[:Actor, :Person], [:Actor, :Animal]]
+
+    # Other valid params for the node_labels option are
+    has_many :in, :actors, type: :ACTS_IN, node_labels: [[:Actor, :Person], :Actor]
+
+    # or
+    has_many :in, :actors, type: :ACTS_IN, node_labels: :Actor
+  end
+
+Note, while the ``label_module`` option requires its params to resolve to ``ActiveLabel`` modules, the ``node_labels``
+option doesn't. The ``node_labels`` option simply matches against the specified labels.
+
+Multiple Conditions
+-------------------
+
+Sometimes you may wish for ``ActiveLabel`` code to be associated with an array of labels, rather than a single label.
+Perhaps the code triggers if `any` label in the array is present, or perhaps it only triggers if `all` labels in the
+array are present.
+
+.. code-block:: ruby
+
+  module Hollywood
+    include Neo4j::ActiveLabel
+
+    self.associated_labels = [:Actor, :Director]
+    self.associated_labels_matcher = :any
+
+    # OR
+
+    self.associated_labels = [:Actor, :Director]
+    self.associated_labels_matcher = :all
+    
+  end
+
+By default, ``self.associated_labels_matcher == :any``
+
+included_if block
+~~~~~~~~~~~~~~~~~
+
+Sometimes conditional functionality is limited to one class, and is simple enough that a full ``ActiveLabel`` module seems like
+overkill. You can make use of ``ActiveNode``'s ``included_if_any`` and ``included_if_all`` methods to specify blocks of code that only
+run if `any` or `all` of the specified labels are present on a node (note, these methods resolve their params to labels,
+rather than ``ActiveLabel`` modules. This means that you can match against an optional label which does not have an associated ``ActiveLabel``
+module).
+
+.. code-block:: ruby
+
+  class Person
+    include Neo4j::ActiveLabel
+
+    # only run if a Person node also has the Actor or Director labels
+    included_if_any :Actor, :Director do
+      property :medium_ego
+    end
+
+    # only run if a Person node also has the Actor AND Director labels
+    included_if_all :Actor, :Director do
+      property :large_ego
+    end
```

Comment:
I'm thinking that maybe this should be refactored as just `included_if() do`. If an argument is an array, then each label in the array must be present. So the above examples could be refactored as:
```
included_if :Actor, :Director do
  property :medium_ego
end

included_if [:Actor, :Director] do
  property :large_ego
end
```
This format would also be more powerful and allow
```
included_if [:Actor, Director], :Animal do
  property :name
end
```

Reply:
For that matter, the `self.associated_labels =` option could similarly be changed to accept a 2 dimensional array and eliminate `self.associated_labels_matcher =`

Reply:
Another issue with any potential `included_if() do` block, I'm not sure how the following should be handled
```
class Person
  include Neo4j::ActiveNode
  scope :actors, ->{ where("#{identity}:Actor") }

  included_if :Actor do
    has_many :out, :acts_in, type: :ACTS_IN, model_class: :Movie
  end
end
```
In the above, devs could
```
movie = Movie.create
actor = Person.create
actor.add_label :Actor

actor.acts_in << movie
```
But I could see a dev also wanting to do something like `Person.actors.acts_in`, which they couldn't do because the `has_many :out, :actors` block would never be evaluated on `Person` (only on instances of person).

Maybe this is fine, and would simply be a known limitation of `included_if()` blocks. Devs could still use the blocks to add properties, define methods, or call custom methods. Or maybe there's another option.

I definitely view anything like `included_if()` as a "nice to have" feature. Maybe it would be more trouble than its worth.

Another note: `ActiveLabel` will, I imagine, make devs want the ability to filter an association proxy by label. Something like `Person.all.has_label(:Actor).acts_in`.